src/components/SearchMap.js

```diff
@@ -1,39 +1,62 @@
-import React, {Component} from 'react';
-import {View, ActivityIndicator, StyleSheet, Platform, Keyboard} from 'react-native';
+// 3rd party libraries - core
+import React from 'react';
+import {View, Text, StyleSheet, ActivityIndicator} from 'react-native';
+import {MapView} from 'expo';
 import {Button} from 'react-native-elements';
+// 3rd party libraries - additional
 
-import {MapView} from 'expo';
+// styles and language
+
+// our components - core
+// our components - additional
 
+const SearchMap = ({mapLoaded, region, updateRegion, sites}) => {
+    const {fillScreen, spinnerContainerStyle} = styles;
 
-class SearchMap extends Component {
-    state = {
-        region: {
-            longitude: -105.727939,
-            latitude: 39.695168,
-            longitudeDelta: 1,
-            latitudeDelta: 1
-        },
-        mapLoaded: false
+    const newRegionIsAcceptable = (newRegion) => {
+        const {longitude, latitude, latitudeDelta, longitudeDelta} = newRegion;
+
+        const notZoomedTooFarOut = latitudeDelta < 60 && longitudeDelta < 60;
+        const isNotSF = (Math.abs(longitude - 122.409) > 1) && (Math.abs(latitude - 37.787) > 1);
+
+        return isNotSF && notZoomedTooFarOut;
```

Comment:
Before we were checking absolute values of the difference from the last lat/long delta and the new one. Instead, I opted for something easier which is to check for the two types of bad data we were getting from the `MapView` component:
 1) Is it trying to zoom out to show the entire US or the world? (In either case, we'd never want to zoom that far out for our app)
 2) Is it trying to re-center around SF? Again, in our case we wouldn't be trying to find camping sites in SF, so we can safely ignore these coords as well.

Reply:
weird!